docs/v1/P4Runtime-Spec.adoc

```diff
@@ -2211,73 +2034,88 @@ set, a single entry will be read.
 Let's look at the counter example in more details. Based on this specification
 document, the C++ server code which processes `CounterEntry` messages may look
 like this:
-~ Begin CPP
+
+[source,c++]
+----
 auto *counter_entry = ...
 if (counter_entry->has_index()) {
   auto index = counter_entry->index().index();
   read_one_entry(counter_entry->id(), index);
 } else {
   read_all_entries(counter_entry->id());
 }
-~ End CPP
+----
 
-1. Reading a single counter entry at index 0 in the counter array with id
+. Reading a single counter entry at index 0 in the counter array with id
 `<id>`:
     * Here is the C++ client code:
-~ Begin CPP
+
+[source,c++]
+----
 p4::v1::CounterEntry entry;
 entry.set_counter_id(<id>);
 entry.mutable_index();
 // The above line sets the index field; it is equivalent to:
 // auto *index = entry.mutable_index();
 // index->set_index(0);
-~ End CPP
+----
+
     * Here is the corresponding Protobuf message in text format:
-~ Begin Prototext
+
+[source,protobuf]
+----
 counter_id: <id>
```

Comment:
The PDF shows a red/brown highlight on the closing angle bracket.

Reply:
I suspect this is because the quoted text is not actually a syntactically legal text protobuf.  The colorizing syntax highlighter doesn't know what to do with it.

Reply:
Perhaps we could change the example to somehow be syntactically legal, but still give the idea that `<id>` is meant to be an arbitrary identifier, but I'm not thinking of any ideas on that off-hand.